packages/models/src/base.ts

```diff
@@ -0,0 +1,77 @@
+import { Knex } from "knex";
+import { knex } from "@/query";
+import { pick } from "lodash";
+
+export class Base<
+  Row extends object,
+  Self extends object,
+  FieldToColumnMap extends Record<keyof Self, keyof Row>,
+> {
+  protected columns: Record<keyof Row, string>;
```

Comment:
I was surprised to know that there is a modifier called `protected`! I justed learned about it today!